Add CraftCMS ^4.0.0 compatibility

jeroenlammerts opened the pull request:
See this as a starting point. Not tested all features.

jeroenlammerts committed:
Add CraftCMS ^4.0.0 compatibility

jeroenlammerts committed:
Fix deprecated methods, added typehints and missing imports

jeroenlammerts commented:
#32

samhibberd commented:
Hi @jeroenlammerts, thanks very much for this, we have actually progressed the c4 upgrade / restructure of our repo locally (to cover for multiple versions) and will look to get it released asap in conjunction with your work.

sfsmfc commented:
Hello @samhibberd. Is there any update available concerning this update for Craft 4? Thank you in advanced.

sfsmfc committed:
update js to show filtering in Craft v4

Fix layout bug in Craft v4 Plugin Preview and js fault in "normal" entry filter view

jeffreyzant committed:
fix spaces -> tabs
